src/dbt_client/dbtCoreIntegration.ts

```diff
@@ -1007,4 +1012,33 @@ export class DBTCoreProjectIntegration
     );
     return result;
   }
+
+  async applyDeferConfig(): Promise<void> {
+    const root = getProjectRelativePath(this.projectRoot);
+    const currentConfig: Record<string, DeferConfig> =
+      this.deferToProdService.getDeferConfigByWorkspace();
+    const {
+      deferToProduction,
+      manifestPathForDeferral,
+      favorState,
+      manifestPathType,
+      dbtCoreIntegrationId,
+    } = currentConfig[root];
+    if (deferToProduction) {
+      const manifestFolder = await this.getDeferManifestPath(
+        manifestPathType,
+        manifestPathForDeferral,
+        dbtCoreIntegrationId,
+      );
+      const manifestPath = path.join(manifestFolder, DBTProject.MANIFEST_FILE);
+      await this.python?.lock<void>(
+        (python) =>
+          python!`project.apply_defer_config(${manifestPath}, ${favorState})`,
+      );
+    } else {
+      await this.python?.lock<void>(
+        (python) => python!`project.clear_defer_config()`,
+      );
+    }
```

Comment:
Can we not simply call `project.rebuildManifest() here?`

Reply:
replacing `clear_defer_config` with `safe_parse_project`. Can't directly used `rebuildManifest()` cause this function itself is getting called in `rebuildManifest()` so that defer config applied on extension start.

Reply:
That can be solved right. Just add the logic to `rebuildManifest` instead of splitting it into multiple functions.